fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -323,6 +339,16 @@ def __init__(
 
         self._has_params = len(params) > 0
 
+        # TODO(anj): Should we conditionally do this only if we have params?
+        # TODO(anj): this won't work for MOE params.
+        self.buffer_size = sum(p.numel() for p in params)
+        self.ssd_buffer_filename = ""
+        if self.ssd_offload:
+            self.ssd_buffer_filename = f"{randint(1, int(10E6))}_rank{self.rank}"
```

Comment:
is the file name always xxxx_rankN, where xxxx is a rand number? That can still collide and it looks bad. Why not use the tempfile module to get a temp file name with proper prefix that is more readable?

Reply:
Also, which dir will this file be in? The offload dir seems need to be an argument that's passed in as well?

Reply:
I haven't decided the way forward here hence I can add a TODO to address this in a follow up PR.